### PR TITLE
Rename Theano to Aesara

### LINKED ISSUE #148: Rename project to Aesara
There's a consensus among the PyMC developers to rename this project to Aesara, after the daughter of Theano and Pythagoras (at least, according to a minor tradition).

I propose the following import bindings, which seem aesthetically pleasing. If there's agreement on these bindings, we might as well set an example and use them in our own imports (e.g. our test suite).

```python
import aesara as ae
import aesara.tensor as aet
```

This will be a large and probably disruptive change:

- [ ] The library itself, but particularly `__init__.py` and `setup.py` must be updated
- [ ] Our test suite will need to use the new name
- [ ] Whatever documentation we have should reference Aesara and not Theano
- [x] The package will have to be re-uploaded to PyPI
- [ ] The GitHub repository itself will also need to be renamed

### PR DESCRIPTION
**Pull request by brandonwillard**
This PR replaces nearly all references to Theano with the new name Aesara.  Some references to old Theano versions and outdated information was also removed. 

Note: we will need to rename this repository before the link changes are valid.

Closes #148.

### CODE REVIEW

**Comment by hvasbath** [comment]
Just out of curiosity and personal interest, what tools do you use to do refactorings/renamings of such huge dimensions?

**Comment by twiecki** [comment]
Well this will be impossible to review, but why 
image?

**Review by michaelosthege — APPROVED** [review]
This PR also closes #283

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,30 +1,30 @@
-If you want to contribute to Theano, have a look at the instructions here:
-http://deeplearning.net/software/theano/dev_start_guide.html
+If you want to contribute to Aesara, have a look at the instructions here:
+https://aesara.readthedocs.io/en/latest/dev_start_guide.html
 
 
-## Migrating PRs from original Theano Repo
-Theano-PyMC is actively merging new changes. If you have a pull request on the original respository and would like to move it here use the following commands in your local theano repo
+## Migrating PRs from original Aesara Repo
```
Comment (OriolAbril) [inline CONTRIBUTING.md:5]: I think this section should either be removed altogether or keep the theano

File: README.rst [hunk 2]
```diff
@@ -4,13 +4,9 @@
 efficiently evaluate mathematical expressions involving multi-dimensional
 arrays.  It can use GPUs and perform efficient symbolic differentiation.
 
-This is a fork of the `original Theano library <https://github.com/Theano/Theano>`__ that is being
+This is a fork of the `original Theano library <https://github.com/pymc-devs/aesara>`__ that is being
```
Comment (OriolAbril) [inline README.rst:7]: This link should still point to original theano repo

File: aesara/breakpoint.py [hunk 3]
```diff
@@ -35,9 +35,9 @@ class PdbBreakpoint(Op):
 
     .. code-block:: python
 
-        import theano
-        import theano.tensor as tt
-        from theano.breakpoint import PdbBreakpoint
+        import aesara
+        import aesara.tensor as tt
```
Comment (OriolAbril) [inline aesara/breakpoint.py:39]: do we want to update these imports in docs to the proposal in #148 `import aesara as ae; import aesara.tensor as aet`?

**Comment by dfm** [comment]
I set up https://aesara.readthedocs.io, but I don't have permissions to setup the GitHub webhook. If someone else has the permissions and wants to add the hook, I can DM you the URL or add you as an admin on RTDs.

**Comment by brandonwillard** [comment]
> Well this will be impossible to review, but why
> image?

Some old files and outdated documentation were removed.

File: aesara/breakpoint.py (hunk 3 above)
Comment (brandonwillard) [reply aesara/breakpoint.py:39]: Yeah, I was thinking of `at`, though, but `ae*` isn't bad.

File: README.rst (hunk 2 above)
Comment (brandonwillard) [reply README.rst:7]: Yes, it should.

File: CONTRIBUTING.md (hunk 1 above)
Comment (brandonwillard) [reply CONTRIBUTING.md:5]: Yes, definitely.

File: .mailmap [hunk 4]
```diff
@@ -1,263 +0,0 @@
...
-# # Name that should be used <email that should be used> Bad name <bad email>
-# #
-# # You can skip Bad name if it is the same as the one that should be used, and is unique.
-# #
-# # This file is up-to-date if the command git log --format="%aN <%aE>" | sort -u
-# # gives no duplicates.
-#     5	Firstname Lastname <firstname.lastname@example.net>
-#     4	Laboratoire d'Informatique des Systemes Adaptatifs <lisa@iro.umontreal.ca>
-#    25	projects@lgcm <projects@lgcm>
-
```
[the remark this answers is not in the document]
Comment (brandonwillard) [reply .mailmap:13]: True; I just don't like the idea of publicly hosting a bunch of spam-able email addresses&mdash;especially when it's unnecessary and really the responsibility of those individual GitHub users.

We could provide a link to [the relevant GitHub documentation](https://docs.github.com/en/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address), but even that seems like overkill.  This is really a user + GitHub issue and I don't feel comfortable using this project's documentation to instruction people on their personal GitHub configurations.

**Comment by brandonwillard** [comment]
> Just out of curiosity and personal interest, what tools do you use to do refactorings/renamings of such huge dimensions?

For this, I used a combination of Vim/Evil's `s//g`, handwritten `libcst` and `rope` scripts, `ag` and `helm-ag`, and `projectile-replace`.  In total, these changes took maybe 25 minutes to formulate and execute, and nearly as long to check/review.

There were a few bugs due to locally uncaught rearranged imports in the `gpuarray` modules, but that's about it.

File: aesara/breakpoint.py (hunk 3 above)
Comment (OriolAbril) [reply aesara/breakpoint.py:39]: It does not need to be decided here nor now, but probably whatever is used in the docstrings (herr and at pymc3) will be what ends up sticking around

**Comment by twiecki** [comment]
tt does not make sense to me, at does.

**Comment by fonnesbeck** [comment]
I think `aet` was suggested on Slack, with general agreement.

**Comment by twiecki** [comment]
I like that even more.

**Comment by OriolAbril** [comment]
aet is also what is proposed in #148 and nobody complained about it in the issue

**Comment by brandonwillard** [comment]
All right, unless there are any other big requests, I'll merge this one once the tests finish/pass.

**Pull request merged**